src/ml/neural_net/image_augmentation.cpp

```diff
@@ -117,5 +117,98 @@ image_augmenter::result resize_only_image_augmenter::prepare_images(
   return res;
 }
 
+image_augmenter::result processed_image_augmenter::prepare_images(
+    std::vector<labeled_image> source_batch) {
+  result res;
+
+  const size_t n = opts_.batch_size;
+  constexpr size_t c = 3;
+
+  std::vector<turi::neural_net::shared_float_array> images_to_aug;
+  std::vector<turi::neural_net::shared_float_array> ann_batch;
+  std::vector<turi::neural_net::shared_float_array> pred_batch;
+
+  // Decode a batch of images to raw format
+  for (size_t i = 0; i < n; i++) {
+    size_t input_height = source_batch[i].image.m_height;
+    size_t input_width = source_batch[i].image.m_width;
+    std::vector<float> img(input_height * input_width * c, 0.f);
+    // unsigned char *outptr = reinterpret_cast<unsigned char *>(img.data());
+    image_util::copy_image_to_memory(source_batch[i].image, img.data(),
+                                     {input_width * c, c, 1},
+                                     {input_height, input_width, c}, true);
+
+    std::transform(img.begin(), img.end(), img.begin(),
+                   [](float pixel) -> float { return pixel / 255; });
+    shared_float_array image_to_aug =
+        shared_float_array::wrap(img, {input_height, input_width, c});
```

Comment:
Let's use `std::move(img)` here instead of `img` to avoid a copy. In fact, I would recommend factoring out a helper function that converts `image_type` to `shared_float_array`, so you can reason more robustly about when `img` becomes moved-from (invalid)